코드 실행 기능을 테스트한다.

this-is-spear opened the pull request:
### Motivation

사용자의 요청이 올바르게 동작하는지 판단하기 위해서 e2e 테스트를 진행합니다.

### Modifications

e2e 테스트를 진행했습니다. 진행한 테스트는 다음과 같습니다.

- 사용자는 알맞은 코드를 실행하면 성공 결과를 반환받습니다.
- 사용자는 결과가 다른 코드를 실행하면 실패 결과를 반환받습니다.
- 사용자는 예외가 발생하는 코드를 실행하면 예외 결과를 반환받습니다.
- 사용자가 결과를 조회합니다.
- 사용자가 결과 리스트를 조회합니다.

> - 통합 테스트에 필요한 데이터를 저장하기 위해 In memory database를 구현했습니다. 
> - 더미 데이터는 DummyDataLoader에서 관리합니다.

### Results

- 코드 실행 요청을 할 수 있습니다.
- 결과 리스트 조회 요청을 할 수 있습니다.
- 결과 조회 요청을 할 수 있습니다.

this-is-spear committed:
Style(executor) 작성 체크 리스트

###Description###
- 작성 테스트 케이스
- 작성 체크 리스트

this-is-spear committed:
Test(executor) 추가 코드 실행 통합 테스트

###Description###
- 추가 통합 테스트
- 수정 메모리 데이터베이스 : 통합 테스트가 성공할 수 있도록 수정

this-is-spear committed:
Test(executor) 통합 테스트 결과 리스트 조회

this-is-spear committed:
Test(executor) 통합 테스트 결과 조회

this-is-spear committed:
Style(tms) : 수정 TMS 체크 리스트

this-is-spear committed:
Refact(executor) : 수정 메모리 데이터베이스

###Description###
- 추가 메모리 데이터베이스 용량 조절 메서드

this-is-spear committed:
Style(executor) : 삭제 관심사 외 코드

f-lab-jason reviewed (approved):
여기도 일단 db 셋업이 되고 나면 바뀔 부분이 많을것 같은데 지금은 이대로도 좋습니다 😃

this-is-spear commented on executor/src/main/java/code/infra/MemoryExecutionResultRepository.java:
```diff
@@ -1,27 +1,83 @@
+        return Mono.fromRunnable(map::clear);
+    }
+
+    private void adjustMaximumCapacity() {
+        if (map.size() > 200) {
+            String id = map.values().stream()
+                           .min(Comparator.comparing(ExecutionResult::getCreatedAt))
+                           .get().getId();
+            map.remove(id);
+        }
     }
```
@f-lab-jason 리뷰 감사합니다! 
`Map`은 일종의 메모리 데이터베이스로 이용하고 있습니다! 말씀하신 `cache`는 데이터베이스를 외부로 분리하면서 `Guava` 라이브러리로 구현해볼 생각입니다! ☺️

말씀하신 `LinkedHashMap` 클래스에는 `Eldest Entry`를 찾는 메서드를 제공하지 않는데, `Deque`를 함께 이용해서 제거를 해줘야하는걸까요? 🥺

this-is-spear replied:
곰곰히 생각해봤는데, Stream API를 이용해 처음 데이터를 찾자 마자 삭제하면 되겠군요!

```java
//...

private final LinkedHashMap<String, ExecutionResult> map = new LinkedHashMap<>();

private void adjustMaximumCapacity() {
  if (map.size() > 200) {
      String id = map.values().stream()
                     .findFirst()
                     .get().getId();
      map.remove(id);
  }
}

```

f-lab-jason replied:
stream API도 괜찮고 iterator를 쓰셔도 되구요. 기존처럼 '모든 element들을 scan해야하는 상황'만 피하면 좋을 것 같아요

this-is-spear committed:
Refact(executor) : 수정 메모리 데이터베이스

###Description###
- 수정 자료 구조 : LinkedHashMap으로 변경
- 수정 용량 조절 메서드 : Eldest Entry 제거하도록 변경
